monkey/common/agent_events/ping_scan_event.py

```diff
@@ -8,7 +9,9 @@ class PingScanEvent(AbstractAgentEvent):
     An event that occurs when the agent performs a ping scan on its network
 
     Attributes:
-        :param scan_data: The data collected from the ping scan
+        :param response_received: Indicates if target responded to the ping
+        :param os: Operating system type determined by ICMP fingerprinting
     """
 
-    scan_data: PingScanData
+    response_received: bool
+    os: Optional[OperatingSystem]
```

Comment:
We also override target to be IPv4Address and not optional. Check again [this ](https://github.com/guardicore/monkey/pull/2358/files#diff-bd90b8fa2e3a4d99c424e645fbf7a65a6ab36b88ce86bbfd9c474c8654803101)